### PR TITLE
NO-JIRA: feat(olm): Set packageserver replicas to 2 for IBMCloudPlatform

### PR DESCRIPTION
**Pull request by hasueki**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/olm/params.go [hunk 1]
```diff
@@ -66,6 +66,10 @@ func NewOperatorLifecycleManagerParams(hcp *hyperv1.HostedControlPlane, releaseI
 	params.PackageServerConfig.SetRestartAnnotation(hcp.ObjectMeta)
 	params.PackageServerConfig.SetDefaultSecurityContext = setDefaultSecurityContext
 
+	if hcp.Spec.Platform.Type == hyperv1.IBMCloudPlatform && hcp.Spec.ControllerAvailabilityPolicy == hyperv1.HighlyAvailable {
```
Comment (rtheis) [inline control-plane-operator/controllers/hostedcontrolplane/olm/params.go:69]: I don't think that this should be restricted to IBM cloud if standard OCP uses 2 replicas.

**Review by rtheis — CHANGES_REQUESTED** [review]

File: control-plane-operator/controllers/hostedcontrolplane/olm/params.go (hunk 1 above)
Comment (rtheis) [reply control-plane-operator/controllers/hostedcontrolplane/olm/params.go:69]: I think https://github.com/openshift/operator-framework-olm/tree/release-4.16/pkg/package-server-manager controls the package server and uses 2 replicas in HA mode.  That would align with what we see in non-HyperShift clusters.

Comment (rtheis) [reply control-plane-operator/controllers/hostedcontrolplane/olm/params.go:69]: After a quick test, I'm wrong and it appears that the OLM operator reconciles the packageserver deployment.

Comment (rtheis) [reply control-plane-operator/controllers/hostedcontrolplane/olm/params.go:69]: And further testing indeed shows that the package server manager reconciles the packageserver csv and the csv controls the number of replicas.  So again, I think 2 replicas is correct.

Comment (hasueki) [reply control-plane-operator/controllers/hostedcontrolplane/olm/params.go:69]: @csrwng @sjenning Do you have any opinions on this? If we think we should set the replicas down to 2 for HyperShift as a whole, I would be happy to make that adjustment.

**Review by rtheis — APPROVED** [review]
/lgtm

**Pull request merged**